test/validation/api/queue/queue.c

```diff
@@ -1326,6 +1326,138 @@ static int check_sched_queue_aggr(void)
 }
 
+static void aggr_queue_test(const odp_event_aggr_capability_t *capa, odp_queue_type_t queue_type)
+{
+	odp_pool_t evv_pool;
+	odp_pool_param_t pool_param;
+	odp_pool_capability_t pool_capa;
+	odp_queue_param_t param;
+	odp_queue_t queue, aggr_queue;
+	odp_event_aggr_config_t aggr;
+	uint32_t max_vectors = 64;
+	uint32_t vector_size = 2;
+	uint32_t uarea_size = 64;
+	uint32_t num_recv = 0;
+	uint32_t num_events;
+	uint64_t timeout = ODP_TIME_MSEC_IN_NS;
+
+	CU_ASSERT_FATAL(odp_pool_capability(&pool_capa) == 0);
+	if (uarea_size > pool_capa.event_vector.max_uarea_size)
+		uarea_size = pool_capa.event_vector.max_uarea_size;
+
+	CU_ASSERT(capa->max_size <= pool_capa.event_vector.max_size);
+
+	if (max_vectors > capa->max_num)
+		max_vectors = capa->max_num;
+
+	if (vector_size < capa->min_size)
+		vector_size = capa->min_size;
+
+	if (timeout > capa->max_tmo_ns)
+		timeout = capa->max_tmo_ns;
+
+	/* Send events to both normal and aggregator queue */
+	num_events = 2 * (max_vectors / vector_size);
+
+	odp_pool_param_init(&pool_param);
+	pool_param.type = ODP_POOL_EVENT_VECTOR;
+	pool_param.event_vector.num = max_vectors;
+	pool_param.event_vector.max_size = vector_size;
+	pool_param.event_vector.uarea_size = uarea_size;
+	evv_pool = odp_pool_create(NULL, &pool_param);
+	CU_ASSERT_FATAL(evv_pool != ODP_POOL_INVALID);
+
+	aggr.pool = evv_pool;
+	aggr.max_size = vector_size;
+	aggr.event_type = ODP_EVENT_BUFFER;
+	aggr.max_tmo_ns = timeout;
+
+	odp_queue_param_init(&param);
+	param.type = queue_type;
+	param.size = num_events;
+	param.num_aggr = 1;
+	param.aggr = &aggr;
+
+	queue = odp_queue_create("queue_test_aggr", &param);
+	CU_ASSERT_FATAL(queue != ODP_QUEUE_INVALID);
+
+	aggr_queue = odp_queue_aggr(queue, 0);
+	CU_ASSERT_FATAL(aggr_queue != ODP_QUEUE_INVALID);
+	CU_ASSERT(odp_queue_type(aggr_queue) == ODP_QUEUE_TYPE_AGGR);
+
+	for (uint32_t i = 0; i < num_events; i++) {
+		odp_queue_t dst_queue = i % 2 ? queue : aggr_queue;
+		odp_buffer_t buf = odp_buffer_alloc(pool);
+
+		CU_ASSERT_FATAL(buf != ODP_BUFFER_INVALID);
+
+		CU_ASSERT_FATAL(odp_queue_enq(dst_queue, odp_buffer_to_event(buf)) == 0);
+	}
+
+	odp_time_wait_ns(timeout);
+
+	while (1) {
+		odp_queue_t src_queue;
+		odp_event_t event;
+
+		if (queue_type == ODP_QUEUE_TYPE_PLAIN) {
+			src_queue = queue;
+			event = odp_queue_deq(queue);
+		} else {
+			event = odp_schedule(&src_queue, odp_schedule_wait_time(timeout));
+		}
+
+		if (event == ODP_EVENT_INVALID)
+			break;
+
+		CU_ASSERT(src_queue == queue);
+
+		if (odp_event_type(event) == ODP_EVENT_VECTOR) {
+			odp_event_t *event_tbl;
+			odp_event_vector_t evv = odp_event_vector_from_event(event);
+			uint32_t num = odp_event_vector_tbl(evv, &event_tbl);
+
+			CU_ASSERT(odp_event_vector_size(evv) == num);
+			CU_ASSERT(num <= vector_size);
+			CU_ASSERT(odp_event_vector_type(evv) == ODP_EVENT_BUFFER);
+
+			if (uarea_size)
+				CU_ASSERT(odp_event_vector_user_area(evv) != NULL);
+			CU_ASSERT(odp_event_vector_user_flag(evv) == 0);
```

Comment:
It could also be checked that odp_event_vector_type() is ODP_EVENT_ANY or ODP_EVENT_PACKET.

Reply:
`odp_event_vector_type()` is checked couple lines above. Currently the check only accepts `ODP_EVENT_BUFFER`, as the aggregator has been configured with that event type.